addons/mail/static/src/models/discuss_public_view/discuss_public_view.js

```diff
@@ -0,0 +1,120 @@
+/** @odoo-module **/
+
+import { attr, one2one } from '@mail/model/model_field';
+import { registerNewModel } from '@mail/model/model_core';
+import { clear, create, link } from '@mail/model/model_field_command';
+
+function factory(dependencies) {
+
+    class DiscussPublicView extends dependencies['mail.model'] {
+
+        //--------------------------------------------------------------------------
+        // Public
+        //--------------------------------------------------------------------------
+
+        /**
+         * Displays the thread view instead of the welcome view.
+         */
+        async switchToThreadView() {
+            this.update({ shouldDisplayWelcomeView: false });
```

Comment:
The variable from the python should be "shouldDisplayWelcomeView**Initially**" and be immutable.

Here you should just replace/clear the `welcomeView` relation according to if the view should exist or not.